src/Cli/dotnet/commands/dotnet-publish/Program.cs

```diff
@@ -23,6 +23,27 @@ namespace Microsoft.DotNet.Tools.Publish
 {
     public class PublishCommand : RestoringCommand
     {
+
+        /// <summary>
+        /// The list of properties that should be forwarded from the publish profile to the publish invocation.
+        /// </summary>
+        /// <remarks>
+        /// While we could forward along every property, the intent of this array is to mimic the behavior of VS,
+        /// whose Publish operation only forwards along a few properties. Of particular interest are properties that
+        /// are set very early on in the build (RID, Configuration, etc).  The remainder will be imported during the
+        /// build via the Microsoft.Net.Sdk.Publish props and targets.
+        /// </remarks>
+        private static string[] PropertiesToForwardFromProfile = new [] {
+            "Configuration",
```

Comment:
You may want to use `MSBuildPropertyNames` here, but it doesn't really matter too much.